pilot/pkg/networking/core/v1alpha3/accesslog.go

```diff
@@ -135,34 +138,91 @@ func (b *AccessLogBuilder) setTCPAccessLog(mesh *meshconfig.MeshConfig, config *
 	}
 }
 
-func (b *AccessLogBuilder) setHTTPAccessLog(mesh *meshconfig.MeshConfig, connectionManager *hcm.HttpConnectionManager) {
-	if mesh.AccessLogFile != "" {
-		connectionManager.AccessLog = append(connectionManager.AccessLog, b.buildFileAccessLog(mesh))
+func buildAccessLogFromTelemetry(mesh *meshconfig.MeshConfig, spec *telemetry.Telemetry, forListener bool) *accesslog.AccessLog {
+	// TODO support multiple
+	accessLogConfig := spec.AccessLogging[0]
```

Comment:
If accessloging is empty then return nil.
Or function singnature should say spec.access logging is non empty.

Reply:
Obvious invalid settings should be prevented by validation webhook